### PR TITLE
[build] Fix base OS compilation issue caused by incompatibility of urllib with requests.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### What I did
Fix SONiC image compilation issue:
```
[2023-05-03T17:54:00.146Z] + sudo cp files/build_templates/docker_image_ctl.j2 ./fs********-mellanox/usr/share/sonic/templates/docker_image_ctl.j2
[2023-05-03T17:54:00.146Z] + sudo LANG=C DOCKER_HOST= ch******** ./fs********-mellanox /usr/local/bin/generate_shutdown_order.py
[2023-05-03T17:54:00.146Z] Traceback (most recent call last):
[2023-05-03T17:54:00.146Z]   File "/usr/local/lib/python3.9/dist-packages/docker/api/client.py", line 214, in _retrieve_server_version
[2023-05-03T17:54:00.146Z]     return self.version(api_version=False)["ApiVersion"]
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/docker/api/daemon.py", line 181, in version
[2023-05-03T17:54:00.147Z]     return self._result(self._get(url), json=True)
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/docker/utils/decorators.py", line 46, in inner
[2023-05-03T17:54:00.147Z]     return f(self, *args, **kwargs)
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/docker/api/client.py", line 237, in _get
[2023-05-03T17:54:00.147Z]     return self.get(url, **self._set_request_timeout(kwargs))
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/requests/sessions.py", line 600, in get
[2023-05-03T17:54:00.147Z]     return self.request("GET", url, **kwargs)
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/requests/sessions.py", line 587, in request
[2023-05-03T17:54:00.147Z]     resp = self.send(prep, **send_kwargs)
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/requests/sessions.py", line 701, in send
[2023-05-03T17:54:00.147Z]     r = adapter.send(request, **kwargs)
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/requests/adapters.py", line 486, in send
[2023-05-03T17:54:00.147Z]     resp = conn.urlopen(
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 790, in urlopen
[2023-05-03T17:54:00.147Z]     response = self._make_request(
[2023-05-03T17:54:00.147Z]   File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 496, in _make_request
[2023-05-03T17:54:00.147Z]     conn.request(
[2023-05-03T17:54:00.147Z] TypeError: request() got an unexpected keyword argument 'chunked'
```

#### How I did it
Pin urllib to version < 2 as it was suggested in https://github.com/docker/docker-py/issues/3113.

#### How to verify it
Run SONiC image compilation.

### CODE REVIEW

**Comment by liushilongbuaa** [comment]
It didn't work.

**Comment by oleksandrivantsiv** [comment]
@liushilongbuaa now it is fixed

**Comment by anamehra** [comment]
We are seeing this failure on 202205 as well, please add the fix there too.
@abdosi , FYI.

**Comment by qiluo-msft** [comment]
I see a new release hours ago: https://github.com/docker/docker-py/releases/tag/6.1.0
Could you check whether it really "Support for requests 2.29.0+ and urllib3 2.x"?

**Comment by oleksandrivantsiv** [comment]
@qiluo-msft requests 2.29.0+ and urllib3 2.x doesn't work in the SONiC environment. I got the same error.
The current fix seems to be the correct one. 
Redis-py fixed the issue in the same way: https://github.com/redis/redis-py/pull/2748

**Comment by liushilongbuaa** [comment]
https://dev.azure.com/mssonic/build/_build/results?buildId=268185&view=results
Yesterdays' PR check passed. Maybe it is fixed?

**Comment by keboliu** [comment]
> https://dev.azure.com/mssonic/build/_build/results?buildId=268185&view=results Yesterdays' PR check passed. Maybe it is fixed?

Hi @liushilongbuaa I still see the same issue in my local build environment.

**Comment by liushilongbuaa** [comment]
It is environment issue. Maybe latest ubuntu:20.04 image in azure updated docker to 6.1.0.
For master branch, maybe update docker to 6.1.0 is better.

File: setup.py [hunk 1]
```diff
@@ -221,6 +221,7 @@
     },
     install_requires=[
         'click==7.0',
+        'urllib3<2',
         'click-log>=0.3.2',
         'docker>=4.4.4',
```
Comment (liushilongbuaa) [inline setup.py:226]: docker>=6.1.1

File: setup.py [hunk 2]
```diff
@@ -221,6 +221,7 @@
     },
     install_requires=[
         'click==7.0',
+        'urllib3<2',
```
Comment (liushilongbuaa) [inline setup.py:224]: we don't need to limit urllib3 now.

**Comment by stephenxs** [comment]
A similar issue is also observed in the swss vstest: https://dev.azure.com/mssonic/be1b070f-be15-4154-aade-b1d3bfb17054/_apis/build/builds/268784/logs/95

File: setup.py (hunk 2 above)
Comment (keboliu) [reply setup.py:224]: Hi @liushilongbuaa  seems it's not fixed, I still see same issue in my local build, and it also failed the checker of PR https://github.com/sonic-net/sonic-swss/pull/2763.

**Review by qiluo-msft — APPROVED** [review]

**Comment by anamehra** [comment]
Hi @abdosi , please help to get this one in 202205.

**Comment by rlhui** [comment]
> Hi @abdosi , please help to get this one in 202205.

@yxieca please help on this one, thanks

**Comment by msosyak** [comment]
@qiluo-msft What is needed to trigger sonic-utilities submodule update in sonic-buildimage repo? Could you help with it, please?

**Comment by yxieca** [comment]
@oleksandrivantsiv @liushilongbuaa do we still need to cherry-pick this change? Do we need this change at all?

**Comment by liushilongbuaa** [comment]
Yes. We need.
Local build maybe fails. But it doesn't affect our pipeline.

**Comment by yxieca** [comment]
> Yes. We need. Local build maybe fails. But it doesn't affect our pipeline.

I thought locking docker to 6.1.1 should have fixed it? @liushilongbuaa

**Comment by liushilongbuaa** [comment]
> > Yes. We need. Local build maybe fails. But it doesn't affect our pipeline.
> 
> I thought locking docker to 6.1.1 should have fixed it? @liushilongbuaa

docker 6.1.1 will fix the issue.
So, Line:225 change to docker>=6.1.0 may be better.

**Comment by yxieca** [comment]
Removed cherry-pick flag according to Shilong's comment.

**Pull request merged**